factory/outportBlockCasterFactory.go

```diff
@@ -0,0 +1,67 @@
+package factory
+
+import (
+	"fmt"
+
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/multiversx/mx-chain-ws-connector-template-go/process"
+)
+
+type (
+	headerType = string
+
+	// HeaderCaster is the interface used for the caster to convert from outport.OutportBlock to
+	// data.ShardOutportBlock or data.MetaOutportBlock
+	HeaderCaster interface {
+		Cast() (proto.Message, error)
+	}
+)
+
+const (
+	header    headerType = "Header"
+	headerV2  headerType = "HeaderV2"
+	metaBlock headerType = "MetaBlock"
+)
```

Comment:
we already have these constants in core repo, we can use them from there

Reply:
changed.